app/services/HashingService.coffee

```diff
@@ -0,0 +1,125 @@
+class HashingService
+
+  @reloadPageWithHash: (board, solutionPlacements, SolutionService) ->
+    unless @checkSolutionPlacements board, solutionPlacements, SolutionService
+      @emptyHash()
+      return false
+    HashingService.setHash board.initialValues, board.goals, solutionPlacements
+
+  @emptyHash: -> window.location.hash = ''
+
+  @setHash: (boardValues, goals, solutionPlacements) ->
+    window.location.hash = @encode boardValues, goals, solutionPlacements
+
+  @encode: (boardValues, goals, solutionPlacements) ->
+    boardValues = (JSON.stringify boardValues).replace(/(\[|\]|"|,|{|})*/g, '')
+
+    length = Math.sqrt boardValues.length
+    for list in [0...solutionPlacements.length]
+      for pos in [0...solutionPlacements[list].length]
+        solutionPlacements[list][pos] =
+          solutionPlacements[list][pos][0] * length +
+            solutionPlacements[list][pos][1]
+
+    btoa(JSON.stringify {b: boardValues, g: goals, p: solutionPlacements})
+
+  @decodeMap: () ->
+    try
+      decoded_s  = atob window.location.hash.substr(1, window.location.hash.length)
+      decoded = JSON.parse decoded_s
```

Comment:
we could just make it one line instead of two. We'll just add a newline halfway through to keep to the 80 char limit. This will also take care of creating a new variable